Sources/DistributedSystem/DistributedSystem.swift

```diff
@@ -20,6 +20,10 @@ import struct Foundation.UUID
 internal import NIOCore
 internal import NIOPosix
 
+#if os(Linux)
+typealias uuid_t = (UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8)
```

Comment:
Maybe some comment why it's needed only for Linux?

Reply:
Well, the point here is I do not know why it is needed on Linux...
The simple answer is the `uuid_t` is typealiased in Foundation on macOS and used as an argument type for `UUID` initializer, but in the same time on Linux, while `uuid_t` is defined in `uuid/uuid.h`, it is not defined in the `Foundation`. Probably it is a bug of the linux port of `Foundation`, but may be there are some other reasons which  are not clear to me now....

Reply:
This explanation looks good enough for me:)
```swift
// For some unknown reason `uuid_t` is not defined in Foundation on Linux
```
 [](edited-with-Axolo)